code/datums/components/limp.dm

```diff
@@ -0,0 +1,75 @@
+/*
+	Limping is a component to be attached to carbons for when they have a disparity in leg speed (such as when one of your legs is grievously wounded), so that one step has
+	a longer delay than the other. This is handled as a component rather than being inherent in bodypart code so that we're not constantly tracking footsteps if we don't need to
+
+	TODO: handle both legs being crippled
+*/
+
+/datum/component/limp
+	dupe_mode = COMPONENT_DUPE_HIGHLANDER
+
+	var/obj/item/bodypart/l_leg/left
+	var/obj/item/bodypart/r_leg/right
+	var/additional_slowdown = 0
+	var/obj/item/bodypart/next_leg
+	var/slowdown_left = 0
+	var/slowdown_right = 0
+
+/datum/component/limp/Initialize()
+	if(!iscarbon(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	var/mob/living/carbon/C = parent
+	testing("[C] is limping")
+
+	left = C.get_bodypart(BODY_ZONE_L_LEG)
+	right = C.get_bodypart(BODY_ZONE_R_LEG)
+	update_limp()
+
+	to_chat(C, "<span class='notice'>You feel a limp!</span>")
+
+
+/datum/component/limp/Destroy(force, silent)
+	left = null
+	right = null
+	to_chat(parent, "<span class='notice'>Your walking becomes more regular.</span>")
```

Comment:
should this only be `if !silent`?

Reply:
Don't use `Destroy` for this kind of feedback, if you need to make a proc that calls `qdel` and offers feedback.
`silent` here is not used for this, and text feedback should happen before.